src/Workspaces/MSBuildTest/NewlyCreatedProjectsFromDotNetNew.cs

```diff
@@ -0,0 +1,264 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Test.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.CodeAnalysis.MSBuild.UnitTests
+{
+    public class NewlyCreatedProjectsFromDotNetNew : MSBuildWorkspaceTestBase
+    {
+        private static readonly string s_globalJsonPath;
+
+        // The Maui templates require additional dotnet workloads to be installed.
+        // Running `dotnet workload restore` will install workloads but may require
+        // admin permissions. In addition a restart may be required after workload
+        // installation.
+        private const bool ExcludeMauiTemplates = true;
+
+        protected ITestOutputHelper TestOutputHelper { get; set; }
+
+        static NewlyCreatedProjectsFromDotNetNew()
+        {
+            // We'll use the same global.json as we use for our own build.
+            s_globalJsonPath = Path.Combine(GetSolutionFolder(), "global.json");
+
+            static string GetSolutionFolder()
+            {
+                // Expected assembly path:
+                //  <solutionFolder>\artifacts\bin\Microsoft.CodeAnalysis.Workspaces.MSBuild.UnitTests\<Configuration>\<TFM>\Microsoft.CodeAnalysis.Workspaces.MSBuild.UnitTests.dll
+                var assemblyLocation = typeof(DotNetSdkMSBuildInstalled).Assembly.Location;
+                var solutionFolder = Directory.GetParent(assemblyLocation)
+                    ?.Parent?.Parent?.Parent?.Parent?.Parent?.FullName;
+                Assumes.NotNull(solutionFolder);
+                return solutionFolder;
+            }
```

Comment:
Should this just look up the directory tree and find the first global.json? Rather than all the Parent hardcoding?

Reply:
The `DotNetSdkInstalled` condition uses this same logic to locate the Roslyn.sln. So, if the concern is about the shape of the artifacts folder, then there will already need to be a code change. If the thought it that we could maybe locate a global.json when running on Helix, an earlier version of this code, that used the DotNetSdkLocator to find the pinned SDK version number, was unable to locate a global.json by walking up the folders. I think I will leave this as-is for now.